src/NUnitFramework/framework/Internal/Commands/TestCommand.cs

```diff
@@ -55,12 +58,23 @@ public TestCommand(Test test)
         public Test Test { get; private set; }
 
         /// <summary>
-        /// Runs the test in a specified context, returning a TestResult.
+        /// Runs the test in a specified context, returning a <see cref="TestResult"/>.
         /// </summary>
-        /// <param name="context">The TestExecutionContext to be used for running the test.</param>
-        /// <returns>A TestResult</returns>
+        /// <param name="context">The <see cref="TestExecutionContext"/> to be used for running the test.</param>
         public abstract TestResult Execute(TestExecutionContext context);
 
+        /// <summary>
+        /// Runs the test in a specified context, either setting <see cref="TestExecutionContext.CurrentResult"/> to the result or handling the exception.
+        /// </summary>
+        /// <param name="context">The <see cref="TestExecutionContext"/> to be used for running the test.</param>
+        /// <exception cref="ArgumentNullException">Thrown when <paramref name="context"/> is <see langword="null"/></exception>
+        public void ExecuteAndSetCurrentResult(TestExecutionContext context)
+        {
+            Guard.ArgumentNotNull(context, nameof(context));
+
+            context.ExecuteWithExceptionHandling(c => c.CurrentResult = Execute(c));
+        }
```

Comment:
To my own eye, this shortens the code but obscures it. Others should weigh in.

Reply:
Shortening is not a goal I have for this. I would like to make it clearer. What would you suggest?

Reply:
Only certain test command's handle results and exceptions though - don't they? Coming to this now, my question is why do some places do `Execute`, and some do `ExecuteAndSetCurrentResult`? I'm not sure how better to achieve that.

Did you intend to make `Execute()` return void here?

Reply:
Execute passes the responsibility of handling exceptions up to the test command above you (or the SimpleWorkItem if you're the top). Replacing `Execute` with `ExecuteAndSetCurrentResult` introduces semantic (error vs failure) changes so it should be a case-by-case look.

`Execute` can't return void without forcing us to rewrite all uses of it, but maybe we should?
`ExecuteAndSetCurrentResult` returns void because the entire system requires you to track the result as a side effect on the context. Returning the result would mislead you into thinking that the system is primarily functional rather than side-effecting.

Reply:
> Execute can't return void without forcing us to rewrite all uses of it, but maybe we should?

Are there any more? I had a quick look - I think that may be a good idea to make the API clearer. 🙂 

Thanks for the explanation.

Reply:
Looking through this again - I think this is probably more confusing trying to work out the diff, then it would be to someone coming fresh to the codebase after the change is in. I like that. 🙂

Reply:
I'll make it protected and see what it would look like to add a commit making everything uniform.

Reply:
This does not play well with `ExpectedExceptionCommand` and I'd rather tackle this in its own PR so long as this PR isn't creating a problem.